pdns/dnsdist.cc

```diff
@@ -721,6 +723,48 @@ shared_ptr<DownstreamState> whashed(const NumberedServerVector& servers, const D
   return valrandom(dq->qname->hash(g_hashperturb), servers, dq);
 }
 
+/*
+ * @todo
+ * - test/benchmark other hashing methods
+ * - test/benchmark adding an avalanche algorithm on hashes
+ * @see https://github.com/haproxy/haproxy/blob/master/doc/internals/hashing.txt
+ */
+
+shared_ptr<DownstreamState> chashed(const NumberedServerVector& servers, const DNSQuestion* dq)
+{
+  std::map<unsigned int, shared_ptr<DownstreamState>> circle = {};
+  unsigned int qhash = dq->qname->hash(g_hashperturb);
+
+  for (const auto& d: servers) {
+    if (d.second->isUp()) {
+      if (d.second->hashes.empty()) {
+        // computes server's points
+        // @todo check if server's weight can change over time
```

Comment:
I don't think so, but even if it does I think it might be cleaner to compute the points at the server's creation and when the weight changes, instead of doing it here?

Reply:
Yes, but I did it there because it prevents unnecessary computation when the policy isn't set to `chashed` and I got the feeling the policy handler could be called even if the different pools were not set to it.

Reply:
Right, I feel pre-computing the points might be worth it since it should happen very rarely but let's not rush that decision.